crates/core/src/transport/peer_connection.rs

```diff
@@ -365,23 +366,93 @@ async fn packet_sending(
         packet_id,
         payload,
         outbound_sym_key,
         confirm_receipt,
-    )?;
-    outbound_packets
-        .send((remote_addr, packet.clone().prepared_send()))
-        .await
-        .map_err(|_| TransportError::ConnectionClosed)?;
-    sent_tracker
-        .lock()
-        .report_sent_packet(packet_id, packet.prepared_send());
+    )? {
+        either::Either::Left(packet) => {
+            outbound_packets
+                .send((remote_addr, packet.clone().prepared_send()))
+                .await
+                .map_err(|_| TransportError::ConnectionClosed)?;
+            sent_tracker
+                .lock()
+                .report_sent_packet(packet_id, packet.prepared_send());
+            Ok(())
+        }
+        either::Either::Right((payload, confirm_receipt)) => {
+            // FIXME: how to split confirm_receipt into multiple msgs efficiently?
+            let mut packets = Vec::with_capacity(8);
+
+            let packet = SymmetricMessage::serialize_msg_to_packet_data(
+                packet_id,
+                payload,
+                outbound_sym_key,
+                vec![],
+            )?;
+
+            split_confirm_receipt(packet_id, confirm_receipt, outbound_sym_key, &mut packets)?;
+
+            // Sent the main packet first
+            outbound_packets
+                .send((remote_addr, packet.clone().prepared_send()))
+                .await
+                .map_err(|_| TransportError::ConnectionClosed)?;
+            sent_tracker
+                .lock()
+                .report_sent_packet(packet_id, packet.prepared_send());
+
+            // send confirm receipt packets in parallel
+            let mut futs = packets
+                .into_iter()
+                .map(|packet| async {
+                    outbound_packets
+                        .send((remote_addr, packet.clone().prepared_send()))
+                        .await
+                        .map_err(|_| TransportError::ConnectionClosed)?;
+                    sent_tracker
+                        .lock()
+                        .report_sent_packet(packet_id, packet.prepared_send());
+                    core::result::Result::<(), TransportError>::Ok(())
+                })
+                .collect::<FuturesUnordered<_>>();
+
+            while futs.next().await.is_some() {}
+            Ok(())
+        }
+    }
+}
+
+fn split_confirm_receipt(
+    packet_id: u32,
+    confirm_receipt: Vec<u32>,
+    outbound_sym_key: &Aes128Gcm,
+    packets: &mut Vec<PacketData<SymmetricAES>>,
+) -> Result<()> {
+    match SymmetricMessage::try_serialize_msg_to_packet_data(
+        packet_id,
+        SymmetricMessagePayload::NoOp,
+        outbound_sym_key,
+        confirm_receipt,
+    )? {
+        either::Either::Left(packet) => {
+            packets.push(packet);
+        }
+        either::Either::Right((_, mut confirm_receipt)) => {
+            let mid = confirm_receipt.len() / 2;
```

Comment:
the optimal packet size is to get a noop packet which contains as many receipts which fit into MAX_DATA_SIZE - overhead of the SymmetricMessage serialization